src/modules/orders/services/order_services.py

```diff
@@ -16,15 +16,15 @@ class OrderService:
     funds for buy orders.
     """
 
-    def __init__(self, price_repository: PolygonPricesRepository | None = None):
-        self.price_repository = price_repository or PolygonPricesRepository()
+    def __init__(self, latest_price_service: LatestPriceService | None = None):
+        self.latest_price_service = latest_price_service or LatestPriceService()
 
-    def _get_current_price(self, ticker: str) -> Decimal:
-        price_summary = self.price_repository.get_price(ticker)
-        if not price_summary:
-            raise ValueError("Cannot fetch price for the given instrument.")
-
-        return price_summary.current_price
+    def _get_current_price(self, ticker: str) -> Decimal | None:
+        prices = self.latest_price_service.get_prices()
+        if ticker in prices:
+            return prices[ticker].close
+        else:
+            return None
```

Comment:
you don't need this else here, there is a return statement in the if

Reply:
I just thought it's more verbose